graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -1,28 +1,38 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from collections import namedtuple
+from typing import NamedTuple, Tuple
 
 from ..ast_manipulation import get_only_query_definition, get_only_selection_from_ast
-from ..cost_estimation.helpers import is_int_field_type, is_uuid4_type
-from ..exceptions import GraphQLError
+from ..cost_estimation.helpers import is_uuid4_type
+from ..cost_estimation.int_value_conversion import field_supports_range_reasoning
+from ..exceptions import GraphQLError, GraphQLPaginationError
 
 
-# The intent to split the query at a certain vertex into a certain number of pages.
-VertexPartition = namedtuple(
-    "VertexPartition",
-    (
-        "query_path",  # Tuple[field name : str] leading to the vertex to be split
-        "number_of_splits",  # The number of subdivisions intended for this vertex
-    ),
-)
+class VertexPartition(NamedTuple):
+    """The intent to split the query at a certain vertex into a certain number of pages."""
```

Comment:
I'm not sure if this represent an intent as much as something else... Do you think there's a more precise word we could use here?

Reply:
I can say `plan` instead. Does that make it clearer?

Reply:
Maybe `partitioning plan`? Does that make sense in this context?

Reply:
split = partition. I could say `Plan to split the query ...` or `Plan to partition the query results by adding filters ...`.